packages/itwinui-react/src/core/Table/Table.test.tsx

```diff
@@ -1981,6 +1981,43 @@ it('should select and filter rows', async () => {
   expect(checkboxInputs[3].checked).toBe(false);
 });
 
+it('should only select main row when subComponent is present', async () => {
+  const logSpy = vitest.spyOn(console, 'log');
+  const columns = [
+    {
+      id: 'name',
+      Header: 'Name',
+      accessor: 'name',
+    },
+    {
+      id: 'description',
+      Header: 'description',
+      accessor: 'description',
+    },
+  ];
+
+  const data = mockedData();
+
+  const { container } = renderComponent({
+    columns,
+    data,
+    isSelectable: true,
+    onSelect: (selectedRows) => console.log(selectedRows),
+  });
```

Comment:
Now I'm not sure that this test is representative of the bug. It needs to use `subComponent`.

Reply:
Right, I did miss `subComponent`. I have made the change in the test.